msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/MzMLIntensityPeaksDecoder.java

```diff
@@ -71,45 +78,51 @@ public static DecodedData createEmpty() {
    * provide another method to parseIndexEntries them as floats. Hopefully some day everything will
    * be in 64 bits anyway.
    *
-   * @param bytesIn Byte array, decoded from a base64 encoded string<br>
+   * @param InputStream, decoded from a base64 encoded string<br>
    *        E.g. like: eNoNxltIkwEYBuAOREZFhrCudGFbbraTU+Zmue...
    * @param lengthIn length of data to be treated as values, i.e. the input array can be longer, the
    *        values to be interpreted must start at offset 0, and this will indicate the length
    * @param precision allowed values: 32 and 64, can be null only if MS-NUMPRESS compression was
    *        applied and is specified in the @{code compressions} enum set.
    * @param numPoints a int.
-   * @param compressions null or MzMLCompressionType#NONE have the same effect. Otherwise the binary
-   *        data will be inflated according to the compression rules.
+   * @param compression null or MzMLCompressionType#NO_COMPRESSION have the same effect. Otherwise
+   *        the binary data will be inflated according to the compression rules.
    * @throws java.util.zip.DataFormatException if any.
    * @throws java.io.IOException if any.
    * @return a {@link io.github.msdk.io.mzml2.MzMLIntensityPeaksDecoder.DecodedData} object.
    * @throws io.github.msdk.MSDKException if any.
    */
-  public static DecodedData decode(byte[] bytesIn, int lengthIn, Integer precision, int numPoints,
-      EnumSet<MzMLBinaryDataInfo.MzMLCompressionType> compressions)
+  public static DecodedData decode(InputStream is, int lengthIn, Integer precision, int numPoints,
+      MzMLBinaryDataInfo.MzMLCompressionType compression)
       throws DataFormatException, IOException, MSDKException {
 
     // for some reason there sometimes might be zero length <peaks> tags
     // (ms2 usually)
     // in this case we just return an empty result
-    if (bytesIn.length == 0 || lengthIn == 0) {
+    if (lengthIn == 0) {
       return DecodedData.createEmpty();
     }
-    if (compressions == null) {
-      compressions = EnumSet.noneOf(MzMLBinaryDataInfo.MzMLCompressionType.class);
+
+    EnumSet<MzMLBinaryDataInfo.MzMLCompressionType> compressions =
+        EnumSet.noneOf(MzMLBinaryDataInfo.MzMLCompressionType.class);
+
+    if (compression == MzMLCompressionType.NUMPRESS_LINPRED_ZLIB) {
+      compressions.add(MzMLCompressionType.NUMPRESS_LINPRED);
+      compressions.add(MzMLCompressionType.ZLIB);
+    } else if (compression == MzMLCompressionType.NUMPRESS_POSINT_ZLIB) {
+      compressions.add(MzMLCompressionType.NUMPRESS_POSINT);
+      compressions.add(MzMLCompressionType.ZLIB);
+    } else if (compression == MzMLCompressionType.NUMPRESS_SHLOGF_ZLIB) {
+      compressions.add(MzMLCompressionType.NUMPRESS_SHLOGF);
+      compressions.add(MzMLCompressionType.ZLIB);
+    } else if (compression == null) {
+      compressions.add(MzMLCompressionType.NO_COMPRESSION);
+    } else {
+      compressions.add(compression);
```

Comment:
This block of code seems repeated in multiple places below. Can you refactor into a single helper method?